torch/csrc/jit/passes/xnnpack_rewrite.cpp

```diff
@@ -296,13 +296,28 @@ void FoldPrePackingOps(script::Module& m) {
   PrePackingOpsFolder(m, filter_fn, "prepack_folding");
 }
 
-c10::optional<script::Module> optimizeForMobile(const script::Module& m) {
+script::Module optimizeForMobile(
+    const script::Module& m,
+    const std::map<MobileOptimizerType, bool>& whitelist_optimizers) {
```

Comment:
Also do we need a bool?  It could as well be const std::vector<MobileOptimizerType> & blacklist.  If it's in the blacklist, then skip, otherwise apply.

Reply:
Maybe `unordered_set` if possible instead of map?

Reply:
The initial idea was by default we run through all the optimization pass, but also let customer to customize it by passing a map where they can assign the value for each pass true/pass. Both of you are right, in that case we could have just created a blacklist vector to make it simpler.